slick/src/main/scala/slick/util/AsyncExecutor.scala

```diff
@@ -56,19 +56,61 @@ object AsyncExecutor extends Logging {
 
     @volatile private[this] var executor: ThreadPoolExecutor = _
 
+    if (maxConnections > maxThreads) {
+      // NOTE: when using transactions or DB locks, it may happen that a task has a lock on the database but no thread
+      // to complete its action, while other tasks may have all the threads but are waiting for the first task to
+      // complete. This creates a deadlock.
+      logger.warn("Having maxConnection > maxThreads can result in deadlocks if transactions or database locks are used.")
+    }
```

Comment:
Are we really sure about this? 
Could you maybe explain in a bit more detail how this can happen?
We should be able to write a test if this is the case.

Reply:
Ok, just read up on some issues and saw this comment https://github.com/slick/slick/issues/1614#issuecomment-284730145. That explains it pretty well.
A warning here is ok!